api-consumption/src/main/java/org/zalando/nakadi/service/subscription/state/StartingState.java

```diff
@@ -38,7 +38,7 @@ public void onEnter() {
      * 4. Switches to streaming state.
      */
     private void registerSessionAndStartStreaming() {
-        final boolean subscriptionJustInitialized = SubscriptionInitializer.initializeSubscriptionLocked(getZk(),
+        final boolean subscriptionJustInitialized = SubscriptionInitializer.initialize(getZk(),
                 getContext().getSubscription(), getContext().getTimelineService(), getContext().getCursorConverter());
         getContext().getCurrentSpan().setTag("session.id", getContext().getSessionId());
         if (!subscriptionJustInitialized) {
```

Comment:
I think that this flag is not needed at all, but I also understand that it is not a part of PR intention. Would be nice to have it removed.